backend/main.py

```diff
@@ -0,0 +1,99 @@
+import sqlite3, uuid, json
+from bottle import Bottle, run, request, response
+
+app = Bottle()
+
+# DB mit SQLite
+con = sqlite3.connect("users.db")
+cur = con.cursor()
+
+def init():
+    createDB()
+
+def createDB():
+    sql_query = """
+    CREATE TABLE IF NOT EXISTS users (
+    id VARCHAR(36),
+    nick VARCHAR(30),
+    sessionid VARCHAR(30)
+    );"""
+    cur.execute(sql_query)
+    con.commit()
+
+def insertUserToDB(nick,sessionid):
+    sql_query = """
+    INSERT INTO users (id,nick,sessionid)
+    VALUES ('"""+str(uuid.uuid4())+"','"+nick+"','"+sessionid+"""');"""
+    cur.execute(sql_query)
+    con.commit()
+
+def queryToJSON(query):
+    re = []
+    for e in query:
+        re.append({"id":e[0],"nick":e[1],"sessionid":e[-1]})
+    return(json.dumps(re))
+
+def getAllUsers():
+    sql_query = """SELECT * FROM users"""
+    cur.execute(sql_query)
+    content=cur.fetchall()
+    erg = {}
+    for e in content:
+        erg[e[0]]= {"nick":e[1],"sessionid":e[-1]}
+    return json.dumps((erg))
+
+def getUsersbyQuery(query):
+    #q = query.split(" ")
+    sql_query = """
```

Comment:
As above, serious danger of SQL-Injection, especially since this functions parameter is the user input from the corresponding api call.

Reply:
also parameterized in the new commit.

Reply:
See above, need the DB-API style of parameter substitution.